File: StudAid/StudAid.Model/Advert.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StudAid.Model
{
    public partial class Advert
    {
        public int AdvertId { get; set; }
     
        public string AdvertName { get; set; }
       
        public string AvailableTime { get; set; }
        public double? Price { get; set; }
        public int? Tutor { get; set; }
        public int? SubjectId { get; set; }
        public int? Date { get; set; }
        public string StateMachine { get; set; }
    }
}

File: stud_aid/lib/models/advert.dart
import 'package:json_annotation/json_annotation.dart';

part 'advert.g.dart';

@JsonSerializable()
class Advert {
  int? advertId;
  String? advertName;
  double? price;
  String? availableTime;
  int? tutor;
  int? subjectId;

  Advert() {}
  factory Advert.fromJson(Map<String, dynamic> json) => _$AdvertFromJson(json);

  Map<String, dynamic> toJson() => _$AdvertToJson(this);
}
